### PR TITLE
chore: fix devServer types

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**
chore

**Did you add tests for your changes?**
NA

**If relevant, did you update the documentation?**
NA

**Summary**
- Updated types as per new dev server options

**Does this PR introduce a breaking change?**
no

**Other information**
Continue https://github.com/webpack/webpack-cli/pull/2641

### CODE REVIEW

File: packages/serve/src/types.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
 export type devServerOptionsType = {
-    bonjour?: boolean;
+    bonjour?: boolean | Record<string, any>;
```
Comment (evenstensberg) [inline packages/serve/src/types.ts:2]: `any` for record type?

Comment (anshumanv) [reply packages/serve/src/types.ts:2]: yep they are from external package so left out as any - https://www.npmjs.com/package/bonjour#publishing

File: packages/serve/src/types.ts [hunk 2]
```diff
@@ -17,8 +18,7 @@ export type devServerOptionsType = {
     onAfterSetupMiddleware?: () => void;
     onBeforeSetupMiddleware?: () => void;
     onListening?: () => void;
-    open?: string | boolean | Record<string, unknown>;
-    openPage?: string | string[];
```
Comment (snitin315) [inline packages/serve/src/types.ts:21]: I think we need `openPage` as currently, we have this object of types as union of v3 and v4.

File: packages/serve/src/types.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
 export type devServerOptionsType = {
-    bonjour?: boolean;
+    bonjour?: boolean | Record<string, any>;
     client?: devServerClientOptions;
     compress?: boolean;
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    dev?: Record<string, any>;
+    dev?: Record<string, any>; // drop in dev-server v3
```
Comment (snitin315) [inline packages/serve/src/types.ts:6]:
```suggestion
    dev?: Record<string, any>; // drop in dev-server v4
```

File: packages/serve/src/types.ts (hunk 2 above)
Comment (anshumanv) [reply packages/serve/src/types.ts:21]: yep lets keep it

File: packages/serve/src/types.ts (hunk 3 above)
Comment (anshumanv) [reply packages/serve/src/types.ts]: done

**Review by snitin315 — APPROVED** [review]
Thanks!

**Review by alexander-akait — COMMENTED** [review]
Let's add comment to suppress warning from ts

**Comment by anshumanv** [comment]
Done @alexander-akait

**Review by alexander-akait — APPROVED** [review]

**Review by snitin315 — APPROVED** [review]

**Pull request merged**